src/Microsoft.AspNetCore.Mvc.Core/Internal/AttributeRoute.cs

```diff
@@ -246,11 +228,66 @@ private class RouteInfo
             
             public int Order { get; set; }
 
-            public string RouteGroup { get; set; }
-
             public string RouteName { get; set; }
 
             public RouteTemplate RouteTemplate { get; set; }
         }
+
+        private class RouteInfoEqualityComparer : IEqualityComparer<RouteInfo>
+        {
+            public static readonly RouteInfoEqualityComparer Instance = new RouteInfoEqualityComparer();
+
+            public bool Equals(RouteInfo x, RouteInfo y)
+            {
+                if (x == null && y == null)
```

Comment:
I guess. Could you use the more it in a more convention way in that case:

```
if (x == null)
{
   return y == null;
}
else if (y == null)
{
  return false;
}
...
```

xors are a bit uncommon in our codebase.